models/DataObject/ClassDefinition.php

```diff
@@ -508,13 +508,46 @@ public function generateClassFiles($generateDefinitionFile = true)
 
             $data = '<?php';
             $data .= "\n\n";
+            $data .= $this->getInfoDocBlock();
+            $data .= "\n\n";
 
             $data .= "\nreturn ".$exportedClass.";\n";
 
             \Pimcore\File::putPhpFile($definitionFile, $data);
         }
     }
 
+    /**
+     * @return string
+     *
+     * @internal
+     */
+    protected function getInfoDocBlock(): string
+    {
+        $cd = '/**' . "\n";
+        $cd .= '* Inheritance: '.($this->getAllowInherit() ? 'yes' : 'no')."\n";
+        $cd .= '* Variants: '.($this->getAllowVariants() ? 'yes' : 'no')."\n";
+
+        if ($this->getDescription()) {
+            $description = str_replace(['/**', '*/', '//'], '', $this->getDescription());
+            $description = str_replace("\n", "\n* ", $description);
+
+            $cd .= '* '.$description."\n";
+        }
```

Comment:
Can we have it properly indented, maybe (at least the part that starts with a `*`)? :)

```suggestion
        $cd .= ' * Inheritance: '.($this->getAllowInherit() ? 'yes' : 'no')."\n";
        $cd .= ' * Variants: '.($this->getAllowVariants() ? 'yes' : 'no')."\n";

        if ($this->getDescription()) {
            $description = str_replace(['/**', '*/', '//'], '', $this->getDescription());
            $description = str_replace("\n", "\n * ", $description);

            $cd .= ' * '.$description."\n";
        }
```

Reply:
I simple restored the old code. But yes we can optimize it :)